src/views/SystemIntake/ContactDetails/index.tsx

```diff
@@ -144,23 +143,60 @@ const ContactDetails = ({ systemIntake }: ContactDetailsProps) => {
     return link;
   })();
 
-  const onSubmit = (values?: ContactDetailsForm) => {
-    if (values) {
-      mutate({
-        variables: {
-          input: {
-            id,
-            ...values,
-            governanceTeams: values.governanceTeams || []
-          }
+  const onSubmit = async (values?: ContactDetailsForm) => {
+    if (!values) return null;
+    const updateSystemIntakeContact = async (
+      type: 'businessOwner' | 'productManager' | 'isso'
+    ) => {
+      if (
+        values[type].euaUserId &&
+        values[type].component &&
+        values[type] !== initialValues[type]
+      ) {
+        if (contacts?.[type].id) {
+          return updateContact({ ...values[type], id: contacts[type].id });
         }
-      });
-    }
+        return createContact(values[type]);
+      }
+      return null;
+    };
+
+    await updateSystemIntakeContact('businessOwner');
```

Comment:
Would it be better to use one of the `Promise` methods to run these three updates concurrently, i.e. `.all()`/`.allSettled()`/`.race()`/`.any()`? (I'm not sure which would be the most appropriate here)

Reply:
Looks like `all` will return on the first reject. Is there any error handling necessary before the following mutate?

Reply:
Using `Promise.all()` as of https://github.com/CMSgov/easi-app/pull/1691/commits/41f1393139cb82028d39b013236e04f9812a8b8b.